### ISSUE TITLE
[TW-1933] Consider Data/TASKDATA when locating sync certificates

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1945]
_Holger Hahn on 2017-11-17T09:55:27Z says:_ 

 ## Context:

Configuration
 File: /taskdata/.taskrc (found), 2589 bytes, mode 100644
 Data: /taskdata (found), dir, mode 40755
 TASKDATA: /taskdata
Server: task.mytaskdserver.se:53589
CA: ca.cert.pem, not readable, 0 bytes
Certificate: client.cert.pem, not readable, 0 bytes
Key: client.key.pem, not readable, 0 bytes
## Current behaviour:

While the certificate/key files are located in the corresponding Data directory (/taskdata/ca.cert.pem, ...), they are not found by taskwarrior. The only way to circumvent this situation is to use absolute paths for the certificate/key config in .taskrc.
## Expected behaviour:

Taskwarrior considers the Data/TASKDATA directories in the search path for locating the sync certificate/key files. Example:

(.taskrc)
taskd.certificate=client.cert.pem
taskd.key=client.key.pem
taskd.ca=ca.cert.pem
1. > TASKDATA=/taskdata task sync
should use the certificate/key files /taskdata/ca.cert.pem...
## Motivation and use case:

I like to sync my .taskrc file between different devices, because it contains specific settings for my personal task management, i.e. context, reports, uda, etc. - which I do not want to configure manually on each device.

The TASKDATA environment variable helps me, to create and maintain a .taskrc file, which is platform/user independent. Except for the certificate/key files, they are not portable currently and need to be configured for each system independently.

Alternative: If one wants to keep the secret files away from the Data, a new Env variable can be introduced, e.g. TASKSYNCCREDS.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2017-11-17T09:55:27Z
Modified: 2017-12-02T19:05:49Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2017-12-02T19:05:49Z says:_ 

 The .rc files will be supporting environment variables soon.  I think this will address the issue.

**Comment by tbabej** [comment]
Implemented via #2558.
